targets/TARGET_NXP/TARGET_MCUXpresso_MCUS/TARGET_IMX/i2c_api.c

```diff
@@ -59,11 +57,24 @@ void i2c_init(i2c_t *obj, PinName sda, PinName scl)
 
 int i2c_start(i2c_t *obj)
 {
-    if (LPI2C_MasterStart(i2c_addrs[obj->instance], 0, kLPI2C_Write) != kStatus_Success) {
-        return 1;
-    }
+    LPI2C_Type *base = i2c_addrs[obj->instance];
+    int status = 0;
 
-    return 0;
+    obj->address_set = 0;
+
+    /* Clear all flags. */
+    LPI2C_MasterClearStatusFlags(base, kLPI2C_MasterEndOfPacketFlag |
+                                       kLPI2C_MasterStopDetectFlag |
+                                       kLPI2C_MasterNackDetectFlag |
+                                       kLPI2C_MasterArbitrationLostFlag |
+                                       kLPI2C_MasterFifoErrFlag |
+                                       kLPI2C_MasterPinLowTimeoutFlag |
+                                       kLPI2C_MasterDataMatchFlag);
+
+    /* Turn off auto-stop option. */
+    base->MCFGR1 &= ~LPI2C_MCFGR1_AUTOSTOP_MASK;
```

Comment:
Is this ever turned on again, or is it simply not needed anymore?

Reply:
It is not needed.